### ISSUE TITLE
Workspace Supplier should be a Factory

### ISSUE THREAD

**Issue by bstopp** [adobe/aio-lib-java#85]
Customers may want to integrate to multiple AIO Workspaces/Projects from AEM. The current implementation is limited to one.

**Comment by bstopp** [comment]
I mean that - Customer's AEM environment may want to connect to multiple AIO Workspaces/Projects. To publish or listen to events in different IMS orgs. With the current setup this isn't possible, as far as i can tell.

**Comment by rperelma** [comment]
This is out of scope for this project. It will have to wait for AEM Eventing project.

**Comment by bstopp** [comment]
I'm not sure I follow. 

This project is the primary library for configuring how an AEM instance connects to, and integrates with,  Adobe IO. The IMS library defines the process for getting JWT tokens. 

Limiting the use of this library to only support connecting with or authenticating to a single IMS Org within AEM means that each AEM instance can only ever connect to a single IMS Org. That's not a reasonable assumption to make.

Also - doesn't the `aem` folder contain all of the AEM Eventing projects?

**Comment by rperelma** [comment]
Yes, this library is how it works today. Tomorrow, once [AEM Eventing project](https://wiki.corp.adobe.com/display/WEM/Eventing+Architecture) completes, it will replace the current approach.

**Comment by bstopp** [comment]
Hm. Let me try re-framing the question: 

How would, in either model, a single AEM instance (using the libraries contained here) authenticate to multiple AIO IMS Orgs?

**Comment by rperelma** [comment]
Once AEM Eventing project is done, this library will not be used for integration.
